From 37ce2e42095ac92035ca2c3bb1b9e1b5c5188d80 Mon Sep 17 00:00:00 2001
From: Mashopy <eliasgheeraert@gmail.com>
Date: Wed, 25 Feb 2026 13:49:12 +0530
Subject: [PATCH 3/3] ImsService: Remove references to TelephonyMetrics

Google deprecated TelephonyMetrics class in Android 16 QPR2 as it was considered legacy code [1].

[1] https://android.googlesource.com/platform/frameworks/opt/telephony/+/a1e87f8d20c88efa8887091059d6abb5b5b2f0d0
---
 .../ims/ril/ImsRILAdapter$RilHandler.smali    | 23 -----
 .../com/mediatek/ims/ril/ImsRILAdapter.smali  | 90 -------------------
 .../mediatek/wfo/ril/MwiRIL$RilHandler.smali  | 25 ------
 smali/com/mediatek/wfo/ril/MwiRIL.smali       | 36 --------
 4 files changed, 174 deletions(-)

diff --git a/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali b/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali
index 13a4123..76fb2ef 100644
--- a/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali
+++ b/smali/com/mediatek/ims/ril/ImsRILAdapter$RilHandler.smali
@@ -1091,27 +1091,6 @@
 
     invoke-virtual {v3}, Landroid/os/Message;->sendToTarget()V
 
-    .line 695
-    iget-object v3, p0, Lcom/mediatek/ims/ril/ImsRILAdapter$RilHandler;->this$0:Lcom/mediatek/ims/ril/ImsRILAdapter;
-
-    invoke-static {v3}, Lcom/mediatek/ims/ril/ImsRILAdapter;->-$$Nest$fgetmMetrics(Lcom/mediatek/ims/ril/ImsRILAdapter;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v3
-
-    iget-object v4, p0, Lcom/mediatek/ims/ril/ImsRILAdapter$RilHandler;->this$0:Lcom/mediatek/ims/ril/ImsRILAdapter;
-
-    iget-object v4, v4, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v4}, Ljava/lang/Integer;->intValue()I
-
-    move-result v4
-
-    iget v5, v1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    iget v6, v1, Lcom/mediatek/ims/ril/RILRequest;->mRequest:I
-
-    invoke-virtual {v3, v4, v5, v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilTimeoutResponse(III)V
-
     .line 698
     .end local v2    # "timeoutResponse":Ljava/lang/Object;
     :cond_6
@@ -1334,8 +1313,6 @@
     :goto_2
     return-void
 
-    nop
-
     :pswitch_data_0
     .packed-switch 0x2
         :pswitch_a
diff --git a/smali/com/mediatek/ims/ril/ImsRILAdapter.smali b/smali/com/mediatek/ims/ril/ImsRILAdapter.smali
index 7c0507b..1637cec 100644
--- a/smali/com/mediatek/ims/ril/ImsRILAdapter.smali
+++ b/smali/com/mediatek/ims/ril/ImsRILAdapter.smali
@@ -184,8 +184,6 @@
 
 .field mLastNITZTimeInfo:[Ljava/lang/Object;
 
-.field private mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
 .field public mModemIndication:Lcom/mediatek/ims/ril/ModemIndication;
 
 .field public mModemResponse:Lcom/mediatek/ims/ril/ModemResponse;
@@ -330,14 +328,6 @@
 
 
 # direct methods
-.method static bridge synthetic -$$Nest$fgetmMetrics(Lcom/mediatek/ims/ril/ImsRILAdapter;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-    .locals 0
-
-    iget-object p0, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    return-object p0
-.end method
-
 .method static bridge synthetic -$$Nest$fgetmMtkRadioVersion(Lcom/mediatek/ims/ril/ImsRILAdapter;)Ljava/util/Map;
     .locals 0
 
@@ -723,13 +713,6 @@
     iput-object v0, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMtkRadioVersion:Ljava/util/Map;
 
     .line 434
-    invoke-static {}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->getInstance()Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v0
-
-    iput-object v0, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    .line 441
     new-instance v0, Lcom/mediatek/ims/ril/ImsRILAdapter$DtmfQueueHandler;
 
     invoke-direct {v0, v1}, Lcom/mediatek/ims/ril/ImsRILAdapter$DtmfQueueHandler;-><init>(Lcom/mediatek/ims/ril/ImsRILAdapter;)V
@@ -5547,27 +5530,6 @@
     .line 4339
     :cond_1
     :goto_0
-    iget-object v1, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v0, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
-
-    move-result v2
-
-    iget v3, p1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    iget v5, p1, Lcom/mediatek/ims/ril/RILRequest;->mRequest:I
-
-    move v4, p2
-
-    move-object v6, p4
-
-    .end local p2    # "rilError":I
-    .end local p4    # "ret":Ljava/lang/Object;
-    .local v4, "rilError":I
-    .local v6, "ret":Ljava/lang/Object;
-    invoke-virtual/range {v1 .. v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilSolicitedResponse(IIIILjava/lang/Object;)V
 
     .line 4342
     iget p2, p1, Lcom/mediatek/ims/ril/RILRequest;->mRequest:I
@@ -8600,19 +8562,6 @@
     iget v2, v1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
 
     invoke-virtual {v0, v2}, Lcom/mediatek/ims/ril/RadioVoiceProxy;->acceptCall(I)V
-
-    .line 1737
-    iget-object v2, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v3, p0, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v3}, Ljava/lang/Integer;->intValue()I
-
-    move-result v3
-
-    iget v4, v1, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    invoke-virtual {v2, v3, v4}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeRilAnswer(II)V
     :try_end_0
     .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
     .catch Ljava/lang/RuntimeException; {:try_start_0 .. :try_end_0} :catch_0
@@ -17738,45 +17687,6 @@
 
     .line 4475
     :goto_0
-    iget-object v13, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v6, v1, Lcom/mediatek/ims/ril/ImsRILAdapter;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v6}, Ljava/lang/Integer;->intValue()I
-
-    move-result v14
-
-    iget v15, v4, Lcom/mediatek/ims/ril/RILRequest;->mSerial:I
-
-    .line 4476
-    invoke-virtual {v0, v2}, Ljava/lang/Object;->equals(Ljava/lang/Object;)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_1
-
-    .line 4477
-    move/from16 v17, v12
-
-    goto :goto_1
-
-    .line 4478
-    :cond_1
-    const/4 v0, 0x2
-
-    move/from16 v17, v0
-
-    :goto_1
-    nop
-
-    .line 4475
-    const/16 v16, 0x3
-
-    const-wide/16 v18, 0x0
-
-    invoke-virtual/range {v13 .. v19}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeRilSendSms(IIIIJ)V
-
-    .line 4483
     goto :goto_2
 
     .line 4472
diff --git a/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali b/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali
index d918894..33c3fde 100644
--- a/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali
+++ b/smali/com/mediatek/wfo/ril/MwiRIL$RilHandler.smali
@@ -193,27 +193,6 @@
 
     invoke-virtual {v1}, Landroid/os/Message;->sendToTarget()V
 
-    .line 406
-    iget-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL$RilHandler;->this$0:Lcom/mediatek/wfo/ril/MwiRIL;
-
-    invoke-static {v1}, Lcom/mediatek/wfo/ril/MwiRIL;->-$$Nest$fgetmMetrics(Lcom/mediatek/wfo/ril/MwiRIL;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v1
-
-    iget-object v4, p0, Lcom/mediatek/wfo/ril/MwiRIL$RilHandler;->this$0:Lcom/mediatek/wfo/ril/MwiRIL;
-
-    iget-object v4, v4, Lcom/mediatek/wfo/ril/MwiRIL;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v4}, Ljava/lang/Integer;->intValue()I
-
-    move-result v4
-
-    iget v5, v2, Lcom/mediatek/wfo/ril/RILRequest;->mSerial:I
-
-    iget v6, v2, Lcom/mediatek/wfo/ril/RILRequest;->mRequest:I
-
-    invoke-virtual {v1, v4, v5, v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilTimeoutResponse(III)V
-
     .line 409
     .end local v3    # "timeoutResponse":Ljava/lang/Object;
     :cond_1
@@ -296,12 +275,8 @@
 
     invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    move-result-object v3
-
     invoke-virtual {v3, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
 
-    move-result-object v3
-
     invoke-virtual {v3}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
 
     move-result-object v3
diff --git a/smali/com/mediatek/wfo/ril/MwiRIL.smali b/smali/com/mediatek/wfo/ril/MwiRIL.smali
index ffb7fb3..d215003 100644
--- a/smali/com/mediatek/wfo/ril/MwiRIL.smali
+++ b/smali/com/mediatek/wfo/ril/MwiRIL.smali
@@ -83,8 +83,6 @@
 
 .field mIsMobileNetworkSupported:Z
 
-.field private mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
 .field final mPhoneId:Ljava/lang/Integer;
 
 .field private mRILDefaultWorkSource:Landroid/os/WorkSource;
@@ -129,14 +127,6 @@
 
 
 # direct methods
-.method static bridge synthetic -$$Nest$fgetmMetrics(Lcom/mediatek/wfo/ril/MwiRIL;)Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-    .locals 0
-
-    iget-object p0, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    return-object p0
-.end method
-
 .method static bridge synthetic -$$Nest$mclearWakeLock(Lcom/mediatek/wfo/ril/MwiRIL;I)Z
     .locals 0
 
@@ -267,13 +257,6 @@
 
     iput-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mTestingEmergencyCall:Ljava/util/concurrent/atomic/AtomicBoolean;
 
-    .line 293
-    invoke-static {}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->getInstance()Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    move-result-object v1
-
-    iput-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
     .line 305
     const/4 v1, 0x0
 
@@ -3464,25 +3447,6 @@
 
     .line 1147
     :goto_0
-    iget-object v1, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mMetrics:Lcom/android/internal/telephony/metrics/TelephonyMetrics;
-
-    iget-object v0, p0, Lcom/mediatek/wfo/ril/MwiRIL;->mPhoneId:Ljava/lang/Integer;
-
-    invoke-virtual {v0}, Ljava/lang/Integer;->intValue()I
-
-    move-result v2
-
-    iget v3, p1, Lcom/mediatek/wfo/ril/RILRequest;->mSerial:I
-
-    iget v4, p2, Landroid/hardware/radio/V1_0/RadioResponseInfo;->error:I
-
-    iget v5, p1, Lcom/mediatek/wfo/ril/RILRequest;->mRequest:I
-
-    move-object v6, p3
-
-    .end local p3    # "ret":Ljava/lang/Object;
-    .local v6, "ret":Ljava/lang/Object;
-    invoke-virtual/range {v1 .. v6}, Lcom/android/internal/telephony/metrics/TelephonyMetrics;->writeOnRilSolicitedResponse(IIIILjava/lang/Object;)V
 
     .line 1149
     if-eqz p1, :cond_2
-- 
2.53.0

